binary_gentoo/internal/atoms.py

```diff
@@ -19,3 +20,10 @@ def extract_category_package_from(atomlike):
     else:
         raise ValueError(f'Not valid "{ATOM_LIKE_DISPLAY}" syntax: {atomlike!r}')
     return match.group('category'), match.group('package')
+
+
+def extract_set_from(atomlike):
+    match = re.compile(_set_pattern).match(atomlike)
+    if match is None:
+        raise ValueError(f'Not valid "{ATOM_LIKE_DISPLAY}" syntax: {atomlike!r}')
```

Comment:
Use of `atomlike` and `ATOM_LIKE_DISPLAY` seems out of place in this function. Let's maybe use a new `SET_DISPLAY` and `set_candidate` or so here. We can catch the ValueError above and throw a new ValueError with a message mention sets and atoms, just an idea.